contracts/commands/ModernTriggerExecutor.sol

```diff
@@ -0,0 +1,183 @@
+     * @return Correctness of the trigger execution
+     */
+    function isExecutionCorrect(bytes memory triggerData) external view override returns (bool) {
+        ModernTriggerData memory modernTriggerData = abi.decode(triggerData, (ModernTriggerData));
+        IVerifier verifier = getVerifier(modernTriggerData.triggerType);
+
+        address[] memory tokens = new address[](2);
+        tokens[0] = modernTriggerData.quoteToken;
+        tokens[1] = modernTriggerData.collateralToken;
+
+        checkIfContractEmpty(tokens);
+        return verifier.isExecutionCorrect(triggerData);
+    }
+
+    /**
+     * @notice Checks the validity of the trigger data when the trigger is created
+     * @param triggerData  Encoded trigger data struct
+     * @return Correctness of the trigger data
+     */
+    function isTriggerDataValid(
+        bool continuous,
+        bytes memory triggerData
+    ) external view override returns (bool) {
+        ModernTriggerData memory modernTriggerData = abi.decode(triggerData, (ModernTriggerData));
+        IVerifier verifier = getVerifier(modernTriggerData.triggerType);
+        return verifier.isTriggerDataValid(continuous, triggerData);
+    }
+
+    /**
+     * @notice Checks the validity of the trigger data when the trigger is executed
+     * @param triggerData  Encoded trigger data struct
+     * @return Correctness of the trigger data during execution
+     */
+    function isExecutionLegal(bytes memory triggerData) external view override returns (bool) {
+        ModernTriggerData memory modernTriggerData = abi.decode(triggerData, (ModernTriggerData));
+        IVerifier verifier = getVerifier(modernTriggerData.triggerType);
+        return verifier.isExecutionLegal(triggerData);
+    }
+
+    /**
+     * @dev Validates the operation hash by decoding the input data and comparing it with the provided operation hash.
+     * @param _data The operation executor execution data containing the operation hash.
+     * @param operationHash The expected operation hash stored in trigger data.
+     */
+    function validateoperationHash(bytes calldata _data, bytes32 operationHash) public pure {
+        (, string memory decodedoperationHash) = abi.decode(_data[4:], (Call[], string));
+        require(
+            keccak256(abi.encodePacked(decodedoperationHash)) == operationHash,
+            "modern-trigger/invalid-operation-name"
+        );
+    }
+
+    /**
+     * @dev Checks if the contract is empty for the given tokens.
+     * @param tokens An array of token addresses to check.
+     * @dev Throws an error if any of the tokens have a non-zero balance in the contract.
+     *      In case of weth it also checks the ETH balance of the contract itself.
+     */
+    function checkIfContractEmpty(address[] memory tokens) public view {
+        for (uint256 i = 0; i < tokens.length; i++) {
+            require(
+                (IERC20(tokens[i]).balanceOf(self) == 0) &&
```

Comment:
I'm concerned about this logic for several reasons:
- This function seems to be an internal functionality (used in the `isExecutionCorrect`) and I'm not sure it should be public
- In general, if possible, `view` functions should not revert, unless they are internal and the revert is part of the internal logic. This should instead return a `bool` that can be checked by the caller and then decide what to do.

If you decide to change it, I'd suggest calling it `isContractEmpty`. If left like it is then I'd make it internal